src/main/java/org/broadinstitute/hellbender/dev/utils/ReadRecord.java

```diff
@@ -0,0 +1,274 @@
+package org.broadinstitute.hellbender.dev.utils;
+
+import com.google.api.services.genomics.model.CigarUnit;
+import com.google.api.services.genomics.model.LinearAlignment;
+import com.google.api.services.genomics.model.Position;
+import com.google.api.services.genomics.model.Read;
+import com.google.common.collect.Iterables;
+
+import htsjdk.samtools.CigarOperator;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SAMTag;
+import htsjdk.samtools.SAMTagUtil;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Read -> SAMRecord.
+ */
+public class ReadRecord extends SAMRecord {
+
+  private enum Tag {
+
+    AM(Type.INTEGER),
+    AS(Type.INTEGER),
+    BC(Type.STRING),
+    BQ(Type.STRING),
+    CC(Type.STRING),
+    CM(Type.INTEGER),
+    CO(Type.STRING),
+    CP(Type.INTEGER),
+    CQ(Type.STRING),
+    CS(Type.STRING),
+    CT(Type.STRING),
+    E2(Type.STRING),
+    FI(Type.INTEGER),
+    FS(Type.STRING),
+    FZ(Type.SHORT_ARRAY),
+    H0(Type.INTEGER),
+    H1(Type.INTEGER),
+    H2(Type.INTEGER),
+    HI(Type.INTEGER),
+    IH(Type.INTEGER),
+    LB(Type.STRING),
+    MC(Type.STRING),
+    MD(Type.STRING),
+    MQ(Type.INTEGER),
+    NH(Type.INTEGER),
+    NM(Type.INTEGER),
+    OC(Type.STRING),
+    OP(Type.INTEGER),
+    OQ(Type.STRING),
+    PG(Type.STRING),
+    PQ(Type.INTEGER),
+    PT(Type.STRING),
+    PU(Type.STRING),
+    Q2(Type.STRING),
+    QT(Type.STRING),
+    R2(Type.STRING),
+    RG(Type.STRING),
+    RT(Type.STRING),
+    SA(Type.STRING),
+    SM(Type.INTEGER),
+    TC(Type.INTEGER),
+    U2(Type.STRING),
+    UQ(Type.INTEGER);
+
+    private enum Type {
+
+      INTEGER {
+        @Override Object convert(List<String> list) {
+          return Integer.valueOf(Integer.parseInt(Iterables.getOnlyElement(list)));
+        }
+      },
+
+      SHORT_ARRAY {
+        @Override Object convert(List<String> list) {
+          int size = list.size();
+          short[] array = new short[size];
+          int i = 0;
+          for (String string : list) {
+            array[i++] = Short.parseShort(string);
+          }
+          return array;
+        }
+      },
+
+      STRING {
+        @Override Object convert(List<String> list) {
+          return Iterables.getOnlyElement(list);
+        }
+      };
+
+      abstract Object convert(List<String> list);
+    }
+
+    private final Type type;
+
+    private Tag(Type type) {
+      this.type = type;
+    }
+
+    Object convert(List<String> list) {
+      return type.convert(list);
+    }
+  }
+
+  private static byte[] integerListToByteArray(List<Integer> list) {
+        int size = list.size();
```

Comment:
please check style.  Here and elsewhere in the file:
- indent by 2 in function body
- closing } in 'if' statements should be lined up with the 'i' in 'if', not indented by 2